### PR TITLE
TypeScript 2.4

### PR DESCRIPTION
**Pull request by kamilkisiela**

### CODE REVIEW

**Comment by kamilkisiela** [comment]
@jbaxleyiii :)

**Comment by jbaxleyiii** [comment]
@kamilkisiela THIS IS
AMAZING

and so are you 💯

File: src/core/ObservableQuery.ts [hunk 1]
```diff
@@ -1,4 +1,8 @@
 import {
+  ExecutionResult,
```
Comment (jbaxleyiii) [inline src/core/ObservableQuery.ts]: Why is this needed?

File: src/transport/batchedNetworkInterface.ts [hunk 2]
```diff
@@ -96,7 +96,7 @@ export class HTTPBatchedNetworkInterface extends BaseNetworkInterface {
             const httpResponse = result as Response;
 
             if (!httpResponse.ok) {
-              return this.applyBatchAfterwares({ responses: [httpResponse], options: batchRequestAndOptions })
+              return this.applyBatchAfterwares({ responses: [httpResponse], options: batchRequestAndOptions.options })
```
Comment (jbaxleyiii) [inline src/transport/batchedNetworkInterface.ts]: yikes! Glad this was caught! Thank you!

**Comment by kamilkisiela** [comment]
@jbaxleyiii OH STOP!

Done
